ui/src/app/pages/data/data-set/data-set-component.tsx

```diff
@@ -1325,38 +1336,47 @@ export const DataSetComponent = fp.flow(withUserProfile(), withCurrentWorkspace(
                             content='Requires Owner or Writer permission' disabled={this.canWrite}>
               {this.isCreatingNewDataset ?
                 <Button style={{marginBottom: '2rem', marginRight: '1rem'}} data-test-id='save-button'
-                        onClick ={() => this.setState({openCreateModal: true})}
+                        onClick ={() => this.setState({modalState: ModalState.Create})}
                         disabled={this.disableSave() || !this.canWrite || !dataSetTouched}>
                   Create Dataset
                 </Button> :
                 <Button style={{marginBottom: '2rem', marginRight: '1rem'}} data-test-id='save-button'
                         onClick ={() => this.saveDataset()}
                         disabled={this.state.savingDataset || this.disableSave() || !this.canWrite || !dataSetTouched}>
-                  Save
+                  Save Dataset
                 </Button>}
             </TooltipTrigger>
 
           <TooltipTrigger data-test-id='export-tooltip'
                           content={exportError}
                           disabled={!exportError}>
             <Button style={{marginBottom: '2rem'}} data-test-id='save-button'
-                    onClick ={() => this.setState({openExportModal: true})}
+                    onClick ={() => this.onClickExport()}
                     disabled={this.disableSave() || !!exportError}>
-              Export
+              Analyze
             </Button>
           </TooltipTrigger>
         </div>
-        {this.state.openCreateModal && <CreateModal entityName='Dataset'
+        {this.state.modalState === ModalState.Create && <CreateModal entityName='Dataset'
```

Comment:
opt nit: this is some deep indentation, maybe wrap after `&&` instead, here and below